Thanks for the fix! Unfortunately it looks like this is currently [breaking the regular smoke tests / helm chart tests](https://github.com/cilium/cilium/actions/runs/11151320809/job/30994464060?pr=35186):

```
$ make -C install/kubernetes
...
Error: 1 chart(s) linted, 1 chart(s) failed
==> Linting ./cilium
Error:  templates/: parse error at (cilium/templates/_helpers.tpl:104): undefined variable "$configmap"

make: *** [Makefile:77: lint] Error 1
make: Leaving directory '/home/runner/work/cilium/cilium/install/kubernetes'
Error: Process completed with exit code 2.
```

--- a/install/kubernetes/cilium/templates/_helpers.tpl
+++ b/install/kubernetes/cilium/templates/_helpers.tpl
@@ -129,9 +129,9 @@ To override the namespace and configMap when using `auto`:
 `.Values.k8sServiceLookupNamespace` and `.Values.k8sServiceLookupConfigMapName`
 */}}
 {{- define "k8sServiceHost" }}
-  {{- if eq .Values.k8sServiceHost "auto" }}
-    {{- $configmapName := default "cluster-info" .Values.k8sServiceLookupConfigMapName }}
-    {{- $configmapNamespace := default "kube-public" .Values.k8sServiceLookupNamespace }}
+  {{- $configmapName := default "cluster-info" .Values.k8sServiceLookupConfigMapName }}
+  {{- $configmapNamespace := default "kube-public" .Values.k8sServiceLookupNamespace }}
+  {{- if and (eq .Values.k8sServiceHost "auto") (lookup "v1" "ConfigMap" $configmapNamespace $configmapName) }}
     {{- $configmap := (lookup "v1" "ConfigMap" $configmapNamespace $configmapName) }}
     {{- $kubeconfig := get $configmap.data "kubeconfig" }}
     {{- $k8sServer := get ($kubeconfig | fromYaml) "clusters" | mustFirst | dig "cluster" "server" "" }}
@@ -149,9 +149,9 @@ To override the namespace and configMap when using `auto`:
 `.Values.k8sServiceLookupNamespace` and `.Values.k8sServiceLookupConfigMapName`
 */}}
 {{- define "k8sServicePort" }}
-  {{- if eq .Values.k8sServiceHost "auto" }}
-    {{- $configmapName := default "cluster-info" .Values.k8sServiceLookupConfigMapName }}
-    {{- $configmapNamespace := default "kube-public" .Values.k8sServiceLookupNamespace }}
+  {{- $configmapName := default "cluster-info" .Values.k8sServiceLookupConfigMapName }}
+  {{- $configmapNamespace := default "kube-public" .Values.k8sServiceLookupNamespace }}
+  {{- if and (eq .Values.k8sServiceHost "auto") (lookup "v1" "ConfigMap" $configmapNamespace $configmapName) }}
     {{- $configmap := (lookup "v1" "ConfigMap" $configmapNamespace $configmapName) }}
     {{- $kubeconfig := get $configmap.data "kubeconfig" }}
     {{- $k8sServer := get ($kubeconfig | fromYaml) "clusters" | mustFirst | dig "cluster" "server" "" }}

